docs/aetest/retry.rst

```diff
@@ -0,0 +1,234 @@
+        @aetest.test
+        def testcase1(self):
+            pass
+
+
+Function Arguments
+------------------
+
+The ``retry`` and ``retry_count`` arguments can be passsed as function
+parameters check for retry and retry count values.
+
+        - ``retry`` (bool) -  To check if the section is being retried. (Optional)
+        - ``retry_count`` (int) - To track the retry count (Optional)
+
+.. code-block:: python
+
+        # retry function arguments
+
+        @aetest.retry()
+        @aetest.test
+        def connect_testcase(self, steps, retry, retry_count):
+            with steps.start('Failure case') as step:
+                if retry and retry_count>1:
+                    step.passed()
+                step.failed()
+
+
+Retry CLI argument
+------------------
+
+Retry can be enabled with default settings using the `\-\-retry` argument.
+
+.. code-block:: text
+
+    pyats run manifest job.tem --retry
+
+To provide setting for retry, you can use one of the following formats:
+
+    1. YAML file
+    2. JSON formatted data
+    3. Key/Value Pairs
+
+The JSON and Key/Value pairs can optionally be Base64 encoded.
+
+YAML file
+~~~~~~~~~
+
+.. code-block:: text
+
+    pyats run manifest job.tem --retry retry.yaml
+
+.. code-block:: yaml
+
+    # retry.yaml
+    sections:
+        - Testcase
+        - Testsection
+    retries: 4
+    retry_wait: 2
+
+The section type mentioned under the ``sections`` key will be retried.
+This will retry testcase and testsection 4 times with a waiting period of 2 seconds.
+If no sections provided then the testcase will be retried by default.
+
+.. code-block:: yaml
+
+    # retry.yaml
+    section_results:
+        - failed
+        - errored
+    retries: 4
+    retry_wait: 2
+
+The section results mentioned under the ``section_results`` key will be retried.
+
+.. code-block:: yaml
+
+    testcases:
+        # Testcase class name
+        test_flaky:
+            # Optionally specify retry count and wait time
+            retries: 3
+            retry_wait: 2
+
+To Enable retry on specific sections, refer the above example.
+
+JSON formatted data
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+    pyats run manifest job.tem --retry \
+    {"sections": ["Cleanupsection", "Testsection"], "retries": 2, "retry_wait": 2}
+
+Key/value Pairs
+~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+    pyats run manifest job.tem --retry retries=3 retry_wait=10
+
+Base64 encoded JSON (or Key/Value pair)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below example base64 encoded JSON string has a value of `{"testcases": {"FlakyTest.test_flaky": {"retries": 3, "retry_wait": 10}}}`
+
+Using Base64 encoding is recommended for JSON strings to avoid problems with
+spacing and/or quote interpretation by the unix shell.
+
+.. code-block:: text
+
+    pyats run manifest job.tem --retry \
+    eyJ0ZXN0Y2FzZXMiOiB7IkZsYWt5VGVzdC50ZXN0X2ZsYWt5IjogeyJyZXRyaWVzIjogMywgInJldHJ5X3dhaXQiOiAxMH19fQo=
+
+
+Example output
```

Comment:
Please add example output for testcase retry

Reply:
updated.